packages/workbox-precaching/src/PrecacheController.ts

```diff
@@ -91,6 +99,25 @@ class PrecacheController {
 
       this._urlsToCacheKeys.set(url, cacheKey);
       this._urlsToCacheModes.set(url, cacheMode);
+
+      if (urlsToWarnAbout.length > 0) {
+        const warningMessage = `Workbox is precaching URLs without revision ` +
+          `info: ${urlsToWarnAbout.join(', ')}\nOnce precached, they ` +
+          `will never be updated.\nThis is only safe for URLs which ` +
+          `contain versioning info (e.g. /v1.0/app.js or /app.1234.js, not ` +
+          `/index.html).\nIf your URLs do contain revision info, pass in ` +
+          `a object with a 'url' property and a 'revision' property set to ` +
+          `null (e.g. {url: '/v1.0/index.js', revision: null}).\nIn a future ` +
+          `release, using revision: null will be required.\nLearn more at ` +
+          `https://bit.ly/wb-precache`;
```

Comment:
I do think this is pretty long. And since you're already linking to a URL, I think we can make it shorter and put this information on the page we link to. What about:

"Workbox is precaching URLs without revision info. This is generally NOT safe. Learn more at https://bit.ly/wb-precache." (and pass the list of URLs as the second argument to `console.warn()`).

Reply:
Yeah, I think you're right. Given that we don't have the right verbiage on https://bit.ly/wb-precache yet, I went long, but we can just get that written up in time for the v5 launch.

I've seen issues with logging "live" variables in the JS console from a service worker, where they show up something generic like `[Array]` after the service worker's process has been killed. I'd rather stringify them first to make sure that the log message is still meaningful after the service worker has stopped.